client/extensions/woocommerce/state/data-layer/request/actions.js

```diff
@@ -10,30 +10,73 @@ import {
 	WOOCOMMERCE_REQUEST,
 } from 'woocommerce/state/action-types';
 
-function _createRequestAction( method, siteId, path, body ) {
-	return {
+function _createRequestAction( method, siteId, path, body, onSuccessAction, onFailureAction ) {
```

Comment:
I took out the if statements. I had them in there to avoid the extra properties, but you're right in that it probably isn't worth the extra code.